packages/@aws-cdk/cfnspec/lib/schema/property.ts

```diff
@@ -222,6 +222,19 @@ export function isPropertyScrutinyType(str: string): str is PropertyScrutinyType
   return (PropertyScrutinyType as any)[str] !== undefined;
 }
 
+const tagPropertyNames = {
+  Tags: "",
+  UserPoolTags: "",
+};
+
+export type TagPropertyName = keyof typeof tagPropertyNames;
```

Comment:
Not sure if there are reasons I can't do this, but it was the cleanest way to check a literal type. I suppose I could go enum?

Reply:
Is JSII complaining about this?

```ts
export type TagPropertyName = 'Tags' | 'UserPoolTags';
```

If that's the case, I can see a slightly cleaner way that doesn't require instantiating an "unused" const:

```ts
interface TagProperty {
  Tags: never;
  UserPoolTags: never;
}

export type TagPropertyName = keyof TagProperty;
```

Reply:
jsii didn't complain locally -- figured the full suite would tell me.

Reply:
I meant complaining about my first snippet:

```ts
export type TagPropertyName = 'Tags' | 'UserPoolTags';
```

Sorry if I wasn't clear. And if you don't get errors building the cognito package, you should be good. 👍